File: leetcode_python/remove_duplicates_from_sorted_array.py
class Solution:
    # @param a list of integers
    # @return an integer
    def removeDuplicates(self, A):
        n = len(A)
        number_of_duplicates = 0
        for ind in range(1, n):
            if A[ind] == A[ind - 1]:
                number_of_duplicates += 1
            elif number_of_duplicates > 0:
                A[ind - number_of_duplicates] = A[ind]
        return n - number_of_duplicates

File: leetcode_cpp/remove_duplicates_from_sorted_array.cpp
class Solution {
public:
    int removeDuplicates(int A[], int n) {
        int number_of_duplicates = 0;
        for (int i = 1; i < n; ++i) {
            if (A[i] == A[i - 1]) {
                number_of_duplicates += 1;
            }
            else if (number_of_duplicates > 0) {
                A[i - number_of_duplicates] = A[i];
            }
        }
        return n - number_of_duplicates;
    }
};
